### PR TITLE
Infrastructure Logging

### PR DESCRIPTION
**Pull request by dzwarg**
Use the webpack infrastructure logging features, which enable users to filter the messages emitted from the plugin during compilation.

### CODE REVIEW

File: e2e/index.js [hunk 1]
```diff
@@ -26,7 +26,10 @@ function createConfig(pluginConfig) {
       path: outputPath,
       filename: `tmp.js`
     },
-    plugins: [new HookShellScriptPlugin(pluginConfig)]
+    plugins: [new HookShellScriptPlugin(pluginConfig)],
+    infrastructureLogging: {
+      level: "none"
+    }
```
Comment (dzwarg) [inline e2e/index.js:32]: Infrastructure log messaging is verified in the other tests, and doesn't need to be emitted here.

File: index.js [hunk 2]
```diff
@@ -67,20 +68,18 @@ class HookShellScriptPlugin {
    * @param {string} msg
    */
   _handleError(msg) {
-    msg = `\n[${NAME}] ${msg}\n`;
     if (!this.watch) {
-      throw new Error(msg);
+      throw new Error(`[${NAME}] ${msg}`);
```
Comment (dzwarg) [inline index.js:72]: Kept `NAME` in the error, but removed from the prefix of the logger message, as Webpack will prepend the message with `[${NAME}] `.

File: test/index.js [hunk 3]
```diff
@@ -23,12 +23,18 @@ const mockHook = (delay = 0) => ({
     this.taps.map(t => setTimeout(() => t.cb(t.name), delay));
   }
 });
+const mockLogger = {
+  info: fake(),
+  error: fake()
+};
```
Comment (dzwarg) [inline test/index.js:29]: These are the mock logging methods to verify, so they are `fake()`s to capture calls and arguments.

File: test/index.js [hunk 4]
```diff
@@ -23,12 +23,18 @@ const mockHook = (delay = 0) => ({
...
+const createLogger = () => mockLogger;
 const mockCompiler = () => ({
   options: { watch: false },
   hooks: {
     beforeRun: mockHook(),
     afterCompile: mockHook()
-  }
+  },
+  getInfrastructureLogger: createLogger
```
Comment (dzwarg) [inline test/index.js:37]: The webpack method of creating a logger is to get one off of the current compiler instance.

https://v4.webpack.js.org/api/plugins/#logging

File: test/index.js [hunk 5]
```diff
@@ -64,20 +68,20 @@ test('it creates a new instance', ({ truthy }) => {
 test.cb('it blows up when a hook does not exist', ({ end, is, truthy }) => {
   const plugin = new HookShellScriptPlugin(testHooks);
   try {
-    plugin.apply({ options: {}, hooks: {} });
+    plugin.apply({ options: {}, hooks: {}, getInfrastructureLogger: createLogger });
   } catch (e) {
     truthy(e instanceof Error);
-    is(e.message, '\n[HookShellScriptPlugin] The hook beforeRun does not exist on the Webpack compiler.\n');
+    is(e.message, '[HookShellScriptPlugin] The hook beforeRun does not exist on the Webpack compiler.');
```
Comment (dzwarg) [inline test/index.js:74]: Only the thrown error message is prefixed with `[HookShellScriptPlugin]`, as webpack prepends all other log entries with that string.

**Comment by drewloomer** [comment]
Thanks for this, @dzwarg! I'll take a look ASAP.

**Review by drewloomer — APPROVED** [review]

**Comment by drewloomer** [comment]
@dzwarg Just pushed a new version to NPM! https://www.npmjs.com/package/hook-shell-script-webpack-plugin
Thanks for your contribution.

**Pull request merged**